cocos/ui/UIEditBox/UIEditBox.cpp

```diff
@@ -52,110 +77,381 @@ EditBox::~EditBox()
 #endif
 }
 
+void EditBox::openKeyboard() const
+{
+    _editBoxImpl->openKeyboard();
+}
 
 void EditBox::touchDownAction(Ref* /*sender*/, TouchEventType controlEvent)
 {
     if (controlEvent == Widget::TouchEventType::ENDED) {
-        _editBoxImpl->openKeyboard();
+        openKeyboard();
     }
 }
 
 EditBox* EditBox::create(const Size& size,
-                         const std::string& normalSprite,
-                        TextureResType texType /*= TextureResType::LOCAL*/)
+                         const std::string& normalImage,
+                         TextureResType texType) {
+    return EditBox::create(size, normalImage, "", "", texType);
+}
+
+EditBox* EditBox::create(const Size& size,
+                         const std::string& normalImage,
+                         const std::string& pressedImage /* = "" */,
+                         const std::string& disabledImage /* = "" */,
+                         TextureResType texType /* = TextureResType::LOCAL */)
 {
     EditBox* pRet = new (std::nothrow) EditBox();
-    
-    if (pRet != nullptr && pRet->initWithSizeAndBackgroundSprite(size, normalSprite, texType))
+    if (pRet != nullptr && pRet->initWithSizeAndTexture(size, normalImage, pressedImage, disabledImage, texType))
     {
         pRet->autorelease();
     }
     else
     {
         CC_SAFE_DELETE(pRet);
     }
-    
+
     return pRet;
 }
-    
-    
-EditBox* EditBox::create(const cocos2d::Size &size, cocos2d::ui::Scale9Sprite *normalSprite, ui::Scale9Sprite* /*pressedSprite*/, Scale9Sprite* /*disabledSprite*/)
+
+
+EditBox* EditBox::create(const cocos2d::Size &size, cocos2d::ui::Scale9Sprite *normalSprite, ui::Scale9Sprite* pressedSprite, Scale9Sprite* disabledSprite)
 {
+#if defined(__GNUC__) && ((__GNUC__ >= 4) || ((__GNUC__ == 3) && (__GNUC_MINOR__ >= 1)))
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#elif _MSC_VER >= 1400 //vs 2005 or higher
+#pragma warning (push)
+#pragma warning (disable: 4996)
+#endif
```

Comment:
why put this macro block inside `EditBox::create` function

Reply:
in order to shut down some compile warnings-uses about DEPRECATED functions